### PR TITLE
[NO-MERGE] PR benchmark playground

### PR DESCRIPTION
**Pull request by CarnaViire**
Dummy PR

### CODE REVIEW

**Comment by CarnaViire** [comment]
/benchmark httpclient-kestrel-configured aspnet-citrine-win libs

**Comment by CarnaViire** [comment]
The error is missing `libgssapi_krb5`:

```
  CMake Error at System.Net.Security.Native/extra_libs.cmake:18 (message):
    Cannot find libgssapi_krb5 and System.Net.Security.Native cannot build
    without it.  Try installing libkrb5-dev (or the appropriate package for
    your platform)
```

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win libs --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
Ok, it seems microbenchmarks are broken on crank side, or, to be precise, 9.0 is not supported there yet:
```
Command:
dotnet publish MicroBenchmarks.csproj ....
....
error NETSDK1005: Assets file 'C:\Users .... MicroBenchmarks\project.assets.json' doesn't have a target for 'net9.0'. Ensure that restore has run and that you have included 'net9.0' in the TargetFrameworks for your project.
```

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win libs --variable filter=*SocketSendReceivePerfTest* --application.framework net8.0

File: src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs [hunk 1]
```diff
@@ -1096,6 +1096,8 @@ public int Send(IList<ArraySegment<byte>> buffers, SocketFlags socketFlags, out
             ValidateBlockingMode();
             if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, $"SRC:{LocalEndPoint} DST:{RemoteEndPoint}");
 
+            System.Threading.Thread.Sleep(1); // (to revert) let's sloooow down
```
Comment (MihaZupan) [inline src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs:1099]: :shipit:

**Comment by CarnaViire** [comment]
Ah, I wasn't able to "hack" BenchmarkDotNet pretending it's net8.0 😅 (it was working in HttpClient benchmarks though 🤣)

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win libs --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
😕😒
```
[06:02:51.712] 'application' failed on agent, stopping...
Job failed at runtime with exit code -532462766:
// Wrong assembly binding redirects for CommandLine, loading it from disk anyway.
[STDERR] Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'CommandLine, Version=2.9.1.0, Culture=neutral, PublicKeyToken=5a870481e358d379'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)
[STDERR] File name: 'CommandLine, Version=2.9.1.0, Culture=neutral, PublicKeyToken=5a870481e358d379'
[STDERR]  ---> System.IO.FileLoadException: The located assembly's manifest definition does not match the assembly reference. (0x80131040)
[STDERR]    at BenchmarkDotNet.ConsoleArguments.ConfigParser.Parse(String[] args, ILogger logger, IConfig globalConfig)
[STDERR]    at BenchmarkDotNet.Running.BenchmarkSwitcher.RunWithDirtyAssemblyResolveHelper(String[] args, IConfig config, Boolean askUserForInput)
[STDERR]    at BenchmarkDotNet.Running.BenchmarkSwitcher.Run(String[] args, IConfig config)
[STDERR]    at MicroBenchmarks.Program.Main(String[] args) in C:\Users\Administrator\AppData\Local\Temp\benchmarks-agent\benchmarks-server-7700\rot5btac.sws\src\benchmarks\micro\Program.cs:line 43
```

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win-ampere libs --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win runtime --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
/benchmark microbenchmarks aspnet-citrine-win libs --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
/benchmark micro aspnet-citrine-win libs --variable filter=*SocketSendReceivePerfTest*

**Comment by CarnaViire** [comment]
The error is:
```
[application] [STDERR] Unhandled exception. System.InvalidOperationException: Benchmark Socket
SendReceivePerfTest.SendAsyncThenReceiveAsync_Task: Job-RFFTPV(PowerPlanMode=00000000-0000-000
0-0000-000000000000, Toolchain=InProcessEmitToolchain, IterationTime=250.0000 ms, MaxIteration
Count=20, MinIterationCount=15, WarmupCount=1) takes too long to run. Prefer to use out-of-pro
cess toolchains for long-running benchmarks.
```
😅
Ok I might need to come up with a less impactful "change" if I want to see the result 🙈

**Pull request closed without merging**